brave/brave-core
# feat(wallet): Group by Filter Option

## Description 
Introduces a `Group by` option to the Portfolio's `Filters and display settings`

Options:
- `None` will not group assets by anything and will display all tokens together like we are used to.
- `Accounts` will group assets by each visible `Account`
- `Networks` will group assets by each visible `Network`
- `Assets` (Note yet implemented, will follow up in a different PR)
https://github.com/brave/brave-browser/issues/30889

note: (Group by `Accounts` will be the default for a new wallet user)


Resolves <https://github.com/brave/brave-browser/issues/30753>



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Go to the `Portfolio` page
2. By default your `Assets` should be grouped by `Accounts` for the first time.
3. Open the `Filters and display settings` and change the `Group by` option to `Networks`, click `Save changes`
4. Your assets should be grouped by `Networks`
5. Open the `Filters and display settings` and enable `Hide small balances`, click `Save changes`
6. `Network` sections with a `0` balance should not be clickable to expand.
7. Open the `Filters and display settings` and change the `Group by` option to `None`, click `Save changes`
8. Your assets should not be grouped by anything.
9. Refresh the browser, your selection should be persisted.

https://github.com/brave/brave-core/assets/40611140/f95edce2-0ce5-48d0-9575-94e23ff956a9
